### PR TITLE
Fix juju plugin

### PR DESCRIPTION
**Pull request by karibou**
Two bugs that seems to have gone unnoticed in earlier versions : 
- check directory existence before using it
- check cmd return code before using its results

### CODE REVIEW

File: sos/plugins/juju.py [hunk 1]
```diff
@@ -55,8 +55,12 @@ class Juju(Plugin, UbuntuPlugin):
...
     def get_deployed_services(self):
         cmd = "juju status --format json"
-        return json_load(
-            self.call_ext_prog(cmd)['output'])['services'].keys()
+        juju_cmd = self.call_ext_prog(cmd)
+        if juju_cmd['status'] == 0:
+            return json_load(
+                juju_cmd['output'])['services'].keys()
+        else:
+            return []
```
Comment (bmr-cymru) [inline sos/plugins/juju.py]: I think it's more readable to write this like:

```python
    if juju_cmd['status'] != 0:
        return []
    # do non-fail stuff
```

**Comment by battlemidget** [comment]
Does this PR still apply?

**Comment by bmr-cymru** [comment]
Merge conflicts:

```
Using index info to reconstruct a base tree...
M	sos/plugins/juju.py
Falling back to patching base and 3-way merge...
Auto-merging sos/plugins/juju.py
CONFLICT (content): Merge conflict in sos/plugins/juju.py
error: Failed to merge in the changes.
Patch failed at 0018 [juju] Get Ready for 2.0 and make it less likely to get too much
The copy of the patch that failed is found in: .git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
```

@karibou is this still relevant? Sorry for parking it for so long - let me know and we can get it in for 3.6 if so.

**Comment by BryanQuigley** [comment]
This can be closed, we're rewriting the Juju plugin around the basic format in #1583

**Pull request closed without merging**